components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/BlobTx.java

```diff
@@ -46,11 +48,17 @@ public BlobTx(final BlobStore blobStore) {
     this.blobStore = checkNotNull(blobStore);
   }
 
-  public BlobRef create(InputStream inputStream, Map<String, String> headers) {
-    Blob blob = blobStore.create(inputStream, headers);
+  public AssetBlob create(final InputStream inputStream,
+                          final Map<String, String> headers,
+                          final Iterable<HashAlgorithm> hashAlgorithms,
+                          final String contentType)
+  {
+    final MultiHashingInputStream hashingStream = new MultiHashingInputStream(hashAlgorithms, inputStream);
+    Blob blob = blobStore.create(hashingStream, headers);
     BlobRef blobRef = new BlobRef("NODE", "STORE", blob.getId().asUniqueString());
```

Comment:
we'll need to address this probably pretty soon... "NODE" and "STORE" aren't useful